pandas/core/indexes/base.py

```diff
@@ -1005,6 +1005,29 @@ def to_series(self, **kwargs):
                       index=self._shallow_copy(),
                       name=self.name)
 
+    def to_frame(self, index=True):
+        """
+        Create a DataFrame with the columns the levels of the Index
+
+        .. versionadded:: 0.21.0
+
+        Parameters
+        ----------
+        index : boolean, default True
+            return the Index as the index
+
+        Returns
+        -------
+        DataFrame : a DataFrame containing the original Index data.
+        """
+
+        from pandas import DataFrame
```

Comment:
this does not do the same thing as ``.to_series()`` where the values and the index are the same.
is there a reason you are doing this?

Reply:
> where the values and the index are the same.

Not sure I get you here.  The implementation I wrote tries to be consistent with what was done with `MultiIndex` by construct `DataFrame` with the data and setting the index if needed.

Reply:
@jreback There is an `index` kwarg, and the default is True (also set index of result of be the same as original index and thus values)

Reply:
ok this is reasonable, side issue this is a little suspect:

```
In [4]: pd.MultiIndex.from_product([range(3),list('ab')], names=['foo', 'bar']).to_frame()
Out[4]: 
        bar  foo
foo bar         
0   a     a    0
    b     b    0
1   a     a    1
    b     b    1
2   a     a    2
    b     b    2

In [5]: pd.MultiIndex.from_product([range(3),list('ab')], names=['foo', 'bar']).to_series()
Out[5]: 
foo  bar
0    a      (0, a)
     b      (0, b)
1    a      (1, a)
     b      (1, b)
2    a      (2, a)
     b      (2, b)
dtype: object
```

Reply:
What is suspect about it? 
The index seems toe same in both examples?